pkg/services/query/models.go

```diff
@@ -32,6 +32,21 @@ func (pr parsedRequest) getFlattenedQueries() []parsedQuery {
 }
 
 func (pr parsedRequest) validateRequest(ctx context.Context) error {
+	refIds := make(map[string]bool)
```

Comment:
I think you might want to implement this in `handleQuerySingleDatasource` instead so that it doesn't apply to `executeConcurrentQueries` (public dashboards) which #59428 tracks. FYI @mmandrus

Reply:
why not leave it here to fix both issues? do public dashboards (`executeConcurrentQueries`) require a different approach?

Reply:
I originally opened a separate issue for public dashboards because I was thinking this validation might take place earlier on in the call chain, either in some middleware or in the API handler. This approach avoids code duplication and kills two birds with one stone, so I'd say go for it.

Reply:
thanks for confirming!